### PR TITLE
Docs (Repo Readme): Update the dgraph-docs repo readme file and PULL_REQUEST_TEMPLATE.md with contributor guidelines

### PR DESCRIPTION
**Pull request by aaroncarey**
This PR updates the dgraph-docs repo with a new readme and pull request template that provides more detailed contributor guidelines, including:
- How to title PRs
- Information to include in the description
- Which branches to target for doc development (use `master` when possible)
- Style guide information
- Minor edits throughout
- Removed outdated "deployment" section

Fixes DGRAPH-2917
Cherry-Pick to all release/* branches.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-
+Title: Please use the following format for your PR title:  topic(area): details
+- The "topic" should be one of the following: Docs, Nav or Chore
+- The "area" is the feature (i.e., "GraphQL"), area of the docs (i.e., "Deployment"), or "Other" (for typo fixes and other bug-fix PRs). 
 Sample Titles:
-feat(Enterprise): Backups can now get credentials from IAM
-fix(Query): Skipping floats that cannot be Marshalled in JSON
-perf: [Breaking] json encoding is now 35% faster if SIMD is present
-chore: all chores/tests will be excluded from the CHANGELOG
+  Docs (GraphQL): Document the @deprecated annotation
```
Comment (danielmai) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: According to the example (`topic(area):`), these samples don't need to be capitalized or have a space in-between:

```
docs(graphql): Document the @deprecated annotation
chore(other): Cherry-pick updates from master to release/v20.11
```

**Review by danielmai — APPROVED** [review]
A few comments. Otherwise, looks good to me.

File: README.md [hunk 2]
```diff
@@ -64,17 +91,14 @@ Pass custom Go-GRPC example to the runnable by passing a `customExampleGoGRPC` t
 {{< /runnable >}}
 ```
 
-We cannot pass multiline string as an argument to a shortcode. Therefore, we
-have to make the whole custom example in a single line string by replacing newlines with `\n`.
+**Note:** Runnable doesn't support passing a multiline string as an argument to a shortcode. Therefore, you have to create the whole custom example in a single line string by replacing newlines with `\n`.
 
 ### Deployment
```
Comment (danielmai) [inline README.md:96]: This Deployment section is obsolete. The docs are deployed via Netlify, so this section is irrelevant—there's no more tmux, SSH, or restarts needed anymore.

**Review by bucanero — COMMENTED** [review]
Once we remove the `### Deployment` section, I think we are good to go 👍 

btw, regarding the style guide: now that we are "open to contributions" I think that we could just make the internal style guide public, and include a link in the Readme.

**Review by MichelDiz — APPROVED** [review]

**Comment by bucanero** [comment]
@aaroncarey you can now link to the Style guide here: https://discuss.dgraph.io/t/developer-documentation-style-guide/10955

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (aaroncarey) [reply .github/PULL_REQUEST_TEMPLATE.md]: Good catch, fixed.

File: README.md (hunk 2 above)
Comment (aaroncarey) [reply README.md]: Removed, thanks!

**Pull request merged**